### PR TITLE
Feat/scout 156 dashboard page

### PR DESCRIPTION
**Pull request by dejwi**
### Task Description
**INTENTED TO USE WITH NEW BACKEND PR**

- dashboard page/modules
- create-translation script upgrade
- added react-countup package

### CODE REVIEW

File: src/pages/dashboard.tsx [hunk 1]
```diff
@@ -1,29 +1,84 @@
...
+  const { data: reports, isLoading: reportsLoading } = useReports({})
+  const { data: userReports, isLoading: userReportsLoading } = useReports({
+    userId
+  })
+  const { data: notes, isLoading: notesLoading } = useNotes({})
+  const { data: userNotes, isLoading: userNotesLoading } = useNotes({
+    userId
+  })
+  const { data: latestReport, isLoading: latestReportLoading } = useReports({
+    limit: 2,
```
Comment (jakubfiglak) [inline src/pages/dashboard.tsx:38]: Why are you fetching 2 reports?

**Review by jakubfiglak — CHANGES_REQUESTED** [review]
Good work, there are some issues though. Please see my comments.

File: src/pages/dashboard.tsx (hunk 1 above)
Comment (dejwi) [reply src/pages/dashboard.tsx:38]: forgot to mention that, backend doesnt allow to fetch only 1:p

**Comment by dejwi** [comment]
yeah that shitty ass components was just my learning stage of mui:D All of them basically use same avatar component so I just extracted it. Btw I don't find using sx prop with more than short styling(especially when long theme props are used) appealing

**Comment by dejwi** [comment]
I see that some types changed, especially from required to not required, for now I did enough to pass build but should I create ticket for overall changes ?

**Comment by jakubfiglak** [comment]
> yeah these shitty ass components were just my learning stage of mui:D All of them basically use same avatar component so I just extracted it. Btw I don't find using sx prop with more than short styling(especially when long theme props are used) appealing

Yeah, it might seem a little hairy but it's a recommended way of adding custom styles to MUI components so we should just stick to it. Especially that we don't apply many styles ourselves.

**Comment by jakubfiglak** [comment]
> I see that some types changed, especially from required to not required, for now I did enough to pass build but should I create ticket for overall changes ?

Nah, if the build is passing, everything should be fine. That's what the workflow is for :D

File: src/pages/dashboard.tsx (hunk 1 above)
Comment (jakubfiglak) [reply src/pages/dashboard.tsx:46]: Oh, it doesn't make any sense, I'll create a ticket for that - will fix FE later

File: src/modules/dashboard/CountCard.tsx [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    </Link>
+  </Card>
+)
+
+const CardContentContainer = styled(CardContent)({
+  height: '100%',
+  display: 'flex',
+  alignItems: 'center',
+  justifyContent: 'space-between',
+})
```
Comment (jakubfiglak) [inline src/modules/dashboard/CountCard.tsx:48]: I'd still use `sx` prop for this, there's really not so many properties. And this is definitely "one-time" styling (it's private to `CountCard` component). But I'll leave it up to you.

File: public/locales/en/common.json [hunk 3]
```diff
@@ -191,6 +191,7 @@
   "YELLOW_CARDS": "Yellow cards",
   "RED_CARD": "Red card",
   "RED_CARDS": "Red cards",
+  "AVG_RATING": "Average rating",
   "ORGANIZATION_SUBSCRIPTION": "Organization subscription",
   "ORGANIZATION_SUBSCRIPTIONS": "Organization subscriptions"
 }
```
Comment (jakubfiglak) [inline public/locales/en/common.json:197]: You might wanna take a look why your editor doesn't add `newline` at the end of json files

File: src/modules/dashboard/StyledAvatar.tsx [hunk 4]
```diff
@@ -0,0 +1,14 @@
+import { Avatar, styled } from '@mui/material'
+
+interface IProps {
+  secondary?: boolean
+}
+export const StyledAvatar = styled(Avatar, {
+  shouldForwardProp: prop => prop !== 'secondary',
```
Comment (jakubfiglak) [inline src/modules/dashboard/StyledAvatar.tsx:7]: Why do we need this crazy thing here? :D

**Review by jakubfiglak — APPROVED** [review]
There's still some comments, I'll leave it up to you if you want to address them or not. Especially this thing with forwarding props on `StyledAvatar` component bugs me. This is a really simple thing and we shouldn't need to do such stuff. Sometimes it's better to repeat yourself (in files you're using `StyledAvatar`) than to add unnecessary layer of complexity.

File: public/locales/en/common.json (hunk 3 above)
Comment (dejwi) [reply public/locales/en/common.json:197]: It adds it when I press CTRL+S by myself, auto save doesnt format + using create-translations doesnt add new line🙆‍♂️🙆‍♂️

File: src/modules/dashboard/StyledAvatar.tsx (hunk 4 above)
Comment (dejwi) [reply src/modules/dashboard/StyledAvatar.tsx:7]: mui things😅

**Pull request merged**